fms/training/plugins.py

```diff
@@ -73,9 +69,16 @@ class InferenceValidator(TrainerPlugin):
     """
 
     def __init__(
-        self, prompt_tokens: List[str], tokenizer, device, steps=None, eos_token=None
+        self,
+        model: nn.Module,
+        prompt_tokens: List[str],
+        tokenizer: BaseTokenizer,
+        device: torch.device | str,
```

Comment:
do we want to force users to use python >= 3.10?
I prefer using `Union`.

Reply:
there are a few other places as well.  `Union` might be better unless we have other core components that does require a python version that is newer than 3.10

Reply:
ah, wasn't aware of the limitation. I can change it to Union here and we can start updating other places as we come across them?

In our mypy tests we use python 3.10, I guess you're saying it would fail with < 3.10 though.

@afrittoli are you aware of anything else we're doing that already requires us to use Python >= 3.10, regardless of if we make this type hint change?

Reply:
yes I believe it is only introduced after 3.10

Reply:
updated